pkg/operator/controller/ingress-routes/controller.go

```diff
@@ -0,0 +1,250 @@
+		return nil, err
+	}
+	return c, nil
+}
+
+// Config holds all the things necessary for the controller to run.
+type Config struct {
+	SecretNamespace string
+}
+
+// reconciler handles the actual ingress reconciliation logic in response to
+// events.
+type reconciler struct {
+	config Config
+	client client.Client
+}
+
+// Reconcile expects request to refer to a ingress in the operator namespace,
+// and will do all the work to ensure the ingress is in the desired state.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Info("reconciling", "request", request)
+
+	// Only proceed if we can get the ingress resource.
+	ingress := &configv1.Ingress{}
+	if err := r.client.Get(context.TODO(), request.NamespacedName, ingress); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("ingress cr not found; reconciliation will be skipped", "request", request)
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get ingress %q: %v", request, err)
+	}
+
+	// Generate namespaceName mapping
+	mapToSecretName := map[types.NamespacedName]string{}
+	for _, componentRoute := range ingress.Spec.ComponentRoutes {
+		mapToSecretName[types.NamespacedName{Namespace: componentRoute.Namespace, Name: componentRoute.Name}] = componentRoute.ServingCertKeyPairSecret.Name
+	}
+
+	// For each Status.ComponentRoute resource, if a Spec.ComponentRoute with a matching
+	// namespace.name touple exists create the role and roleBinding for each consumer users.
+	// If no matching Spec.ComponentRoute exists, make sure any RBAC that was generated is removed.
+	for _, componentRoute := range ingress.Status.ComponentRoutes {
+		secretName, ok := mapToSecretName[types.NamespacedName{Namespace: componentRoute.Namespace, Name: componentRoute.Name}]
+
+		// Delete the generated role and roleBinding if they exists for a component not defined in the spec.
+		if !ok {
+			if err := r.deleteServiceKeyPairSecretRole(componentRoute.Name); err != nil {
+				return reconcile.Result{Requeue: true}, err
+			}
+
+			if err := r.deleteServiceKeyPairSecretRoleBinding(componentRoute.Name); err != nil {
+				return reconcile.Result{Requeue: true}, err
+			}
+
+			continue
+		}
+
+		if err := r.ensureServiceCertKeyPairSecretRole(ingress, componentRoute.Name, secretName); err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
+
+		if err := r.ensureServiceCertKeyPairSecretRoleBinding(ingress, componentRoute.Name, componentRoute.ConsumingUsers); err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) ensureServiceCertKeyPairSecretRole(owner metav1.Object, crName, secretName string) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			// TODO: Generate Name and add labels to avoid componentRoute.Name collisions.
+			Name:      crName,
+			Namespace: r.config.SecretNamespace,
+			OwnerReferences: []metav1.OwnerReference{
```

Comment:
No.  I don't see a need or desire to tie this to a cluster resource that can never be deleted.